asmcomp/reloadgen.mli

```diff
@@ -25,3 +25,5 @@ class reload_generic : object
   method fundecl : Mach.fundecl -> Mach.fundecl * bool
     (* The entry point *)
 end
+
+val num_stack_slots: int array
```

Comment:
I considered, but it would be the only result of allocate_registers that is returned directly. Other inputs/outputs are communicated via Reg.t.interf and Interval/Interf. 

Putting num_stack_slots in Reloadgen is not entirely contrary to this PR's goal if we think of Asmgen.regalloc as a single pass. At the end of each iteration, Reloadgen uses   num_stack_slots to create the new Mach.fundecl, without the need to know which register allocation algorithm was used and where to find these results.   It's also nice that we don't need to duplicate this array in coloring/linscan and keep the copies in sync. 
The downside of having num_stack_slots in reloadgen is that coloring/linscan now depend on it.